pkg/codegen/go/gen.go

```diff
@@ -1682,7 +1708,11 @@ func (pkg *pkgContext) genResource(w io.Writer, r *schema.Resource, generateReso
 	fmt.Fprintf(w, "\tpulumi.RegisterOutputType(%sOutput{})\n", name)
 	for _, method := range r.Methods {
 		if method.Function.Outputs != nil {
-			fmt.Fprintf(w, "\tpulumi.RegisterOutputType(%s%sResultOutput{})\n", name, Title(method.Name))
+			if len(method.Function.Outputs.Properties) == 1 {
```

Comment:
One of the existing tests failed because it's missing checking `pkg.liftSingleValueMethodReturns` here.

```
        	            	Diff:
        	            	--- Expected
        	            	+++ Actual
        	            	@@ -178,4 +178,4 @@
        	            	 	pulumi.RegisterOutputType(FooOutput{})
        	            	-	pulumi.RegisterOutputType(FooBarResultOutput{})
        	            	-	pulumi.RegisterOutputType(FooGenerateKubeconfigResultOutput{})
        	            	+	pulumi.RegisterOutputType(fooBarResultOutput{})
        	            	+	pulumi.RegisterOutputType(fooGenerateKubeconfigResultOutput{})
        	            	 }
        	Test:       	TestGeneratePackage/Simple_schema_with_methods
        	Messages:   	example/foo.go
Error:     helpers.go:121: example/foo.go did not agree
    --- FAIL: TestGeneratePackage/Simple_schema_with_methods (0.05s)
```